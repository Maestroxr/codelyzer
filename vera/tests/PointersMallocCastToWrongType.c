
int main() 
{
	int returnValue = 0;
	
	/*OK*/
	int* p1 = (int*)malloc(sizeof(int));
	
	int*** p2 = (int***)malloc(sizeof(int)),** p3 = (int**)malloc(sizeof(int));
	
	if (!p1 || !p2)
	{
		x = -1;
	}
	
	p2 = (int***)malloc(sizeof(int));
	
	p1 = (int*)malloc(sizeof(int));
	
	
	
	/*--------*/
	
	/*PROBLEM*/
	
	int* problem1 = (char*)malloc(sizeof(int));
	int** problem2 = 0;

	
	
	problem2 = (void**)malloc(sizeof(int));
	
	
	
	int*** problem3 = (int*)malloc(sizeof(int));
	if (!problem3)
	{
		x = -1;
	}
	problem3 = (int)malloc(sizeof(int));
	
	
	int**** problem4 = (void****)malloc(sizeof(int)),***** problem5 = (int******)malloc(sizeof(int));
	

	/*********/
	
	if (!p1 || !p2 || !p3 || !problem1 || !problem2 || !problem3 || !problem4 || !problem5)
	{
		x = -1;
	}

	
}


