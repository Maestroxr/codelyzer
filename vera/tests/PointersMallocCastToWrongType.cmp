C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:26: Malloc's cast is to incorrect type, pointer 'problem1' initialized as 'int*' but cast was to 'char*'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:31: Malloc's cast is to incorrect type, pointer 'problem2' initialized as 'int**' but cast was to 'void**'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:35: Malloc's cast is to incorrect type, pointer 'problem3' initialized as 'int***' but cast was to 'int*'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:40: Malloc's cast is to incorrect type, pointer 'problem3' initialized as 'int***' but cast was to 'int'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:43: Malloc's cast is to incorrect type, pointer 'problem5' initialized as 'int*****' but cast was to 'int******'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:43: Malloc's cast is to incorrect type, pointer 'problem4' initialized as 'int****' but cast was to 'void****'.