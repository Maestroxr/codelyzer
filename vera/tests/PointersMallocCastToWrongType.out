NEXT RECURSIVE> value:) line:7 column:34
NEXT> value:int line:7 column:31
NEXT> value:* line:7 column:15
NEXT> value:int line:7 column:12
NEXT RECURSIVE> value:) line:9 column:73
NEXT> value:int line:9 column:70
NEXT> value:* line:9 column:54
NEXT> value:* line:9 column:53
NEXT> value:int line:9 column:50
NEXT RECURSIVE> value:) line:9 column:38
NEXT> value:int line:9 column:35
NEXT> value:* line:9 column:19
NEXT> value:* line:9 column:18
NEXT> value:* line:9 column:17
NEXT> value:int line:9 column:14
NEXT> value:p2 line:11 column:13
NEXT> value:! line:11 column:12
NEXT> value:p1 line:11 column:6
NEXT> value:! line:11 column:5
NEXT RECURSIVE> value:) line:16 column:31
NEXT> value:int line:16 column:28
NEXT> value:* line:16 column:12
NEXT> value:* line:16 column:11
NEXT> value:* line:16 column:10
NEXT> value:int line:16 column:7
NEXT RECURSIVE> value:) line:18 column:29
NEXT> value:int line:18 column:26
NEXT> value:* line:18 column:10
NEXT> value:int line:18 column:7
NEXT RECURSIVE> value:) line:26 column:41
NEXT> value:int line:26 column:38
NEXT> value:* line:26 column:22
NEXT> value:char line:26 column:18
NEXT RECURSIVE> value:) line:31 column:37
NEXT> value:int line:31 column:34
NEXT> value:* line:31 column:18
NEXT> value:* line:31 column:17
NEXT> value:void line:31 column:13
NEXT RECURSIVE> value:) line:35 column:42
NEXT> value:int line:35 column:39
NEXT> value:* line:35 column:23
NEXT> value:int line:35 column:20
NEXT> value:problem3 line:36 column:6
NEXT> value:! line:36 column:5
NEXT RECURSIVE> value:) line:40 column:34
NEXT> value:int line:40 column:31
NEXT> value:int line:40 column:13
NEXT RECURSIVE> value:) line:43 column:95
NEXT> value:int line:43 column:92
NEXT> value:* line:43 column:76
NEXT> value:* line:43 column:75
NEXT> value:* line:43 column:74
NEXT> value:* line:43 column:73
NEXT> value:* line:43 column:72
NEXT> value:* line:43 column:71
NEXT> value:int line:43 column:68
NEXT RECURSIVE> value:) line:43 column:47
NEXT> value:int line:43 column:44
NEXT> value:* line:43 column:28
NEXT> value:* line:43 column:27
NEXT> value:* line:43 column:26
NEXT> value:* line:43 column:25
NEXT> value:void line:43 column:21
NEXT> value:problem5 line:48 column:79
NEXT> value:! line:48 column:78
NEXT> value:problem4 line:48 column:66
NEXT> value:! line:48 column:65
NEXT> value:problem3 line:48 column:53
NEXT> value:! line:48 column:52
NEXT> value:problem2 line:48 column:40
NEXT> value:! line:48 column:39
NEXT> value:problem1 line:48 column:27
NEXT> value:! line:48 column:26
NEXT> value:p3 line:48 column:20
NEXT> value:! line:48 column:19
NEXT> value:p2 line:48 column:13
NEXT> value:! line:48 column:12
NEXT> value:p1 line:48 column:6
NEXT> value:! line:48 column:5
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:26: Malloc's cast is to incorrect type, pointer 'problem1' initialized as 'int*' but cast was to 'char*'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:31: Malloc's cast is to incorrect type, pointer 'problem2' initialized as 'int**' but cast was to 'void**'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:35: Malloc's cast is to incorrect type, pointer 'problem3' initialized as 'int***' but cast was to 'int*'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:40: Malloc's cast is to incorrect type, pointer 'problem3' initialized as 'int***' but cast was to 'int'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:43: Malloc's cast is to incorrect type, pointer 'problem5' initialized as 'int*****' but cast was to 'int******'.
C:\Work\AutoAssessment\codelyzer\vera\tests\PointersMallocCastToWrongType.c:43: Malloc's cast is to incorrect type, pointer 'problem4' initialized as 'int****' but cast was to 'void****'.
